pkg/tecdsa/common/common_test.go

```diff
@@ -0,0 +1,454 @@
+	memberID := group.MemberIndex(2)
+
+	tssPartyID := NewTssPartyIDFromMemberID(memberID)
+
+	testutils.AssertStringsEqual(
+		t,
+		"ID of the TSS party ID",
+		tssPartyID.Id,
+		strconv.Itoa(int(memberID)),
+	)
+
+	testutils.AssertBytesEqual(
+		t,
+		tssPartyID.Key,
+		big.NewInt(int64(memberID)).Bytes(),
+	)
+
+	testutils.AssertStringsEqual(
+		t,
+		"moniker of the TSS party ID",
+		tssPartyID.Moniker,
+		fmt.Sprintf("member-%v", memberID),
+	)
+
+	testutils.AssertIntsEqual(
+		t,
+		"index of the TSS party ID",
+		-1,
+		tssPartyID.Index,
+	)
+}
+
+func TestMemberIDToTssPartyIDKey(t *testing.T) {
+	memberID := group.MemberIndex(2)
+
+	key := MemberIDToTssPartyIDKey(memberID)
+
+	testutils.AssertBigIntsEqual(
+		t,
+		"key of the TSS party ID",
+		big.NewInt(int64(memberID)),
+		key,
+	)
+}
+
+func TestTssPartyIDToMemberID(t *testing.T) {
+	partyID := tss.NewPartyID("2", "member-2", big.NewInt(2))
+
+	memberID := TssPartyIDToMemberID(partyID)
+
+	testutils.AssertIntsEqual(t, "member ID", 2, int(memberID))
+}
+
+func TestResolveSortedTssPartyID(t *testing.T) {
+	groupTssPartiesIDs := []*tss.PartyID{
+		tss.NewPartyID("1", "member-1", big.NewInt(1)),
+		tss.NewPartyID("2", "member-2", big.NewInt(2)),
+		tss.NewPartyID("3", "member-3", big.NewInt(3)),
+		tss.NewPartyID("4", "member-4", big.NewInt(4)),
+		tss.NewPartyID("5", "member-5", big.NewInt(5)),
+	}
+
+	tssParameters := tss.NewParameters(
+		tss.EC(),
+		tss.NewPeerContext(tss.SortPartyIDs(groupTssPartiesIDs)),
+		groupTssPartiesIDs[0],
+		len(groupTssPartiesIDs),
+		2,
+	)
+
+	memberID := group.MemberIndex(2)
+
+	tssPartyID := ResolveSortedTssPartyID(tssParameters, memberID)
+
+	testutils.AssertStringsEqual(
+		t,
+		"ID of the TSS party ID",
+		tssPartyID.Id,
+		strconv.Itoa(int(memberID)),
+	)
+
+	testutils.AssertBytesEqual(
+		t,
+		tssPartyID.Key,
+		big.NewInt(int64(memberID)).Bytes(),
+	)
+
+	testutils.AssertStringsEqual(
+		t,
+		"moniker of the TSS party ID",
+		tssPartyID.Moniker,
+		fmt.Sprintf("member-%v", memberID),
+	)
+
+	testutils.AssertIntsEqual(
+		t,
+		"index of the TSS party ID",
+		int(memberID-1),
+		tssPartyID.Index,
+	)
+}
+
+func TestAggregateTssMessages(t *testing.T) {
+	var tests = map[string]struct{
+		tssMessages              []tss.Message
+		symmetricKeys            map[group.MemberIndex]ephemeral.SymmetricKey
+		expectedBroadcastPayload []byte
+		expectedPeersPayload     map[group.MemberIndex][]byte
+		expectedErr              error
+	}{
+		"happy path": {
+			tssMessages: []tss.Message {
+				newMockTssMessage([]byte{0x02}, group.MemberIndex(2)),
+				newMockTssMessage([]byte{0xAA}), // broadcast message
+				newMockTssMessage([]byte{0x01}, group.MemberIndex(1)),
+				newMockTssMessage([]byte{0x03}, group.MemberIndex(3)),
+			},
+			symmetricKeys: map[group.MemberIndex]ephemeral.SymmetricKey{
+				1: &mockSymmetricKey{[]byte{0x01}},
+				2: &mockSymmetricKey{[]byte{0x02}},
+				3: &mockSymmetricKey{[]byte{0x03}},
```

Comment:
We could use `0x0A`, `0x0B`, `0x0C` as keys so that we could differentiate the message payload from the key used by the `mockSymmetricKey`:

```
1: {0x01, 0xFF, 0x0A},
2: {0x02, 0xFF, 0x0B},
3: {0x03, 0xFF, 0x0C},
```

vs
```
1: {0x01, 0xFF, 0x01},
2: {0x02, 0xFF, 0x02},
3: {0x03, 0xFF, 0x03},
```

Reply:
See d708c42f